packages/itwinui-react/src/core/Toast/Toast.tsx

```diff
@@ -233,7 +243,10 @@ export const Toast = (props: ToastProps) => {
 export type ToastPresentationProps = Omit<
   ToastProps,
   'duration' | 'id' | 'isVisible' | 'onRemove'
-> & { onClose?: () => void } & CommonProps;
+> & {
+  onClose?: () => void;
+  contentProps?: React.ComponentProps<'div'>;
+} & CommonProps;
```

Comment:
Since `ToastPresentationProps` includes `ToastProps`, we can use the `ToastProps.domProps` instead of defining `contentProps` as a new prop.

I also realized that `CommonProps` would be redundant, since `domProps.toastProps` already includes the common props. So we can remove `CommonProps` and use `domProps.toastProps` instead. Removal is possible here since `ToastPresentation` is private.

```suggestion
export type ToastPresentationProps = Omit<
  ToastProps,
  'duration' | 'id' | 'isVisible' | 'onRemove'
> & { onClose?: () => void };
```

Reply:
what's the benefit of this? top-level `className` makes more sense than `domProps.toastProps.className`

i believe AppUI is using `ToastPresentation` (even though it's private) so they might be relying on top-level props

Reply:
> what's the benefit of this? top-level `className` makes more sense than `domProps.toastProps.className`

I feel `domProps` should be the single source of truth, despite the long name (`domProps.toastProps.className`).

Moreover, I saw a TODO saying that we should remove `CommonProps` once it is not used any more. So I believe we want to move away from `CommonProps`.

https://github.com/iTwin/iTwinUI/blob/f79ed42611492393bdaca91aefc9fc592386a19d/packages/itwinui-react/src/core/utils/props.ts#L7-L12

> i believe AppUI is using `ToastPresentation` (even though it's private) so they might be relying on top-level props

Since they are using `<ToastPresentation>`, we can consider keeping `CommonProps` and className. But ideally, we should be free to make breaking changes to our private components/APIs.

Reply:
The only reason we have `domProps` is because toaster's API is imperative; we don't want a hundred different props directly under the options passed to `toaster.positive()`. For actual components, top-level props makes sense.

CommonProps is ok to remove. I'm fine with it, we can forward ref, spread top-level rest props and merge `className`.
```tsx
export type ToastPresentationProps = Omit<
  ToastProps,
  'duration' | 'id' | 'isVisible' | 'onRemove' | 'domProps'
> & {
  onClose?: () => void;
  contentProps?: React.ComponentProps<'div'>;
};

export const ToastPresentation = React.forwardRef((props, forwardedRef) => {
  // ...
}) as PolymorphicForwardRefComponent<'div', ToastPresentationProps>;
```

But you're right about our ability to make changes to private apis, so i'll leave it on @gretanausedaite to decide.

Reply:
I'd like to keep top-level props as it makes the most sense for me.

I'd like this PR to be customisation changes only to keep it granular. Agree with switching to poly component in a different PR as it does not affect props I added.

Reply:
could you just change the props type then? i believe `domProps` needs to be omitted

Reply:
yup, updated

Reply:
The type `ToastPresentationProps` seem to have some type issues. Like when calling `<ToastPresentation>`, is is called with `{...domProps?.toastProps}` of type `React.ComponentProps<'div'>`.

```tsx
<ToastPresentation
  // ...
  {...domProps?.toastProps} // toastProps is of type React.ComponentProps<'div'>;
  contentProps={domProps?.contentProps}
/>
```

But `ToastPresentationProps` does not accept `React.ComponentProps<'div'>`.

Reply:
ah yeah, my suggestion about Polymorphic above should fix this type error

Reply:
#1519

Reply:
There still seems to be some type error when calling `<ToastPresentation>` with `{...domProps?.toastProps}`. It complains about a missing `as`. The error goes away after removing `{...domProps?.toastProps}` or keeping it but adding `as` to `<ToastPresentation>`

![image](https://github.com/iTwin/iTwinUI/assets/45748283/d2760cbf-9185-4b9c-9c3c-8bcb6e108248)

Reply:
`as='div'` should fix it